src/witan/workspace/util.clj

```diff
@@ -26,60 +15,10 @@
            ~@(interleave (repeat name) (map pstep (partition 2 clauses)))]
        ~name)))
 
-(defn iso-date-time-now
-  "The current date and time in iso-date-time format"
-  []
-  (->> (java.util.Date.)
-       (json/generate-string)
-       (drop 1)
-       (butlast)
-       (apply str)))
-
-(defn replacer
-  "Calls  replacement function on different types"
-  [rfn x]
-  (condp = (type x)
-    clojure.lang.Keyword (-> x name rfn keyword)
-    clojure.lang.MapEntry (update x 0 (partial replacer rfn))
-    clojure.lang.PersistentArrayMap (map (partial replacer rfn) x)
-    java.lang.String (rfn x)))
-
-(defn underscore->hyphen
-  "Converts underscores to hyphens"
-  [x]
-  (replacer #(clojure.string/replace % #"_" "-") x))
-
-(defn hyphen->underscore
-  "Convers hyphens to underscores"
-  [x]
-  (replacer #(clojure.string/replace % #"-" "_") x))
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; Custom coercion func
-
-(defn uuid-coercion
-  [schema]
-  (when (= schema java.util.UUID)
-    (fn [x]
-      (java.util.UUID/fromString x))))
-
-(defn workflow-node-coercion
-  [schema]
-  (when (= schema was/WorkflowNode)
-    (fn [x]
-      (cond
-        (and
-         (vector? x)
-         (every? string? x)) (mapv keyword x)
-        (and
-         (vector? x)
-         (vector? (last x))) (-> x
-                                 (update 0 keyword)
-                                 (update 1 (partial mapv keyword)))
-        :else nil))))
-
-(defn param-coercion-matcher
-  [schema]
-  (or (coerce/json-coercion-matcher schema)
-      (workflow-node-coercion schema)
-      (uuid-coercion schema)))
+(defn timestamp
+  ([fmt time]
+   (tf/unparse (tf/formatters fmt) time))
```

Comment:
There's a horrible outstanding java bug around synchronizing time formatters. Wrap this in a little memoized function and things will flow a bit easier!